crates/autopilot/src/run_loop.rs

```diff
@@ -418,6 +419,8 @@ impl RunLoop {
             },
             solutions: solutions
                 .iter()
+                // reverse as solver competition table is sorted from worst to best
+                .rev()
```

Comment:
It is probably worth explaining why it is crucial not to alter this behavior.

Reply:
Given that the ordering is only relevant to keep the DB in a consistent state I would consider moving the reversal and comment into the `persistence` module to completely remove sorting details from the domain.

Reply:
Let's do this as part of https://github.com/cowprotocol/services/issues/3021 as the whole `postprocessing` function will be refactored to resolve this issue.